Simplified/Adobe DRM/AdobeDRMContainer.mm

```diff
@@ -0,0 +1,156 @@
+            BOOL isSevereEpubError = NO;
+            if (err.is_spec_error()
+                    && (err.severity() == ePub3::ViolationSeverity::Critical
+                    || err.severity() == ePub3::ViolationSeverity::Major))
+                isSevereEpubError = YES;
+
+            return ePub3::DefaultErrorHandler(err);
+        };
+        ePub3::SetErrorHandler(sdkErrorHandler);
+
+        ePub3::InitializeSdk();
+        ePub3::PopulateFilterManager();
+        ePub3::AdeptFilter::Register();
+
+        try {
+            container = ePub3::Container::OpenContainer(path.UTF8String);
+            
+        }
+        catch (std::exception& e) { // includes ePub3::ContentModuleException
+            auto msg = e.what();
+            std::cout << msg << std::endl;
+        }
+        catch (...) {
+        }
+        
+        if (container == nullptr) {
+            return nil;
+        }
+
+        package = container->DefaultPackage().get();
+        ePub3::string s = package->TableOfContents()->SourceHref();
+        manifestItem = package->ManifestItemAtRelativePath(s);
+}
+    return self;
+}
+
+- (void *)getDecodedByteStream:(void *)currentByteStream isRangeRequest:(BOOL)isRangeRequest {
+        // Get the manifest item initWithURL: saves for further use
+        ePub3::ManifestItemPtr m = std::const_pointer_cast<ePub3::ManifestItem>(manifestItem);
+        size_t numFilters = package->GetFilterChainSize(m);
+        ePub3::ByteStream *byteStream = nullptr;
+        ePub3::SeekableByteStream *rawInput = (ePub3::SeekableByteStream *)currentByteStream;
+        
+        if (numFilters == 0)
+        {
+            byteStream = (ePub3::ByteStream *) currentByteStream; // is actually a SeekableByteStream
+        }
+        else if (numFilters == 1 && isRangeRequest)
+        {
+            byteStream = package->GetFilterChainByteStreamRange(m, rawInput).release(); // is *not* a SeekableByteStream, but wraps one
+            if (byteStream == nullptr)
+            {
+                byteStream = package->GetFilterChainByteStream(m, rawInput).release(); // is *not* a SeekableByteStream, but wraps one
+            }
+        }
+        else
+        {
+            byteStream = package->GetFilterChainByteStream(m, rawInput).release(); // is *not* a SeekableByteStream, but wraps one
+        }
+        
+        return byteStream;
+
+}
+
+
+- (NSData *)decodeData:(NSData *)data {
+    
+    @synchronized (acsdrm_lock) {
+
+        NSData *m_data;
+        NSUInteger m_contentLength;
+        NSUInteger m_contentLengthCheck;
+        UInt8 m_buffer[1024 * 256];
+        std::unique_ptr<ePub3::ByteStream> m_byteStream;
+        DataByteStream *dataByteStream;
+        
+        dataByteStream = new DataByteStream((unsigned char *)[data bytes], (unsigned long)data.length);
+        
+        if (m_data == nil) {
+            m_byteStream.reset((ePub3::ByteStream *)[self getDecodedByteStream:dataByteStream isRangeRequest:NO]);
```

Comment:
is `m_byteStream` just a pointer on the `data` bytes, essentially?

Reply:
not exactly, `m_byteStream` is a ByteStream, and behaves as a stream - it provides methods to check if the stream is open read/write data.this is not an array of bytes. internal functions need SeekableByteStream (that what the new DataByteStream is), when the stream comes through the filters, we get a decrypted stream back and read it as a normal ByteStream

Reply:
hmm I think I understand... would be good to add this to the in-code comments

Reply:
this code had changed - ePub3 is removed and there're comments in the new code